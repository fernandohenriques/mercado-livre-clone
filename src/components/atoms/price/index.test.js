import React from 'react';
import { mount } from 'enzyme';
import { render } from '@testing-library/react';
import Price from './index';

describe('<Price />', () => {
  it('Snapshot testing with medium size', () => {
    const { asFragment } = render(
      <Price amount={2999} decimals={9} size="medium" />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('Snapshot testing with large size', () => {
    const { asFragment } = render(
      <Price amount={145} decimals={50} size="large" />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('Should use class medium if dont pass props size', () => {
    const wrapper = mount(<Price amount={200} decimals={99} />);

    expect(wrapper.find('.price').hasClass('medium')).toBe(true);
  });
});
