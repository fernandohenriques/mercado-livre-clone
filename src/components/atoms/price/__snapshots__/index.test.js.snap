// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Price /> Snapshot testing with large size 1`] = `
<DocumentFragment>
  <span
    class="price large"
  >
    <span
      class="amount"
    >
      $ 145
    </span>
    <span
      class="decimals"
    >
      50
    </span>
  </span>
</DocumentFragment>
`;

exports[`<Price /> Snapshot testing with medium size 1`] = `
<DocumentFragment>
  <span
    class="price medium"
  >
    <span
      class="amount"
    >
      $ 2.999
    </span>
    <span
      class="decimals"
    >
      09
    </span>
  </span>
</DocumentFragment>
`;
